### PR TITLE
sb3: disable health check for neo4j + reactive

### PR DESCRIPTION
**Pull request by mshima**
---

Please make sure the below checklist is followed for Pull Requests.

- [ ] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [ ] Tests are added where necessary
- [ ] The JDL part is updated if necessary
- [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) is updated if necessary
- [ ] Documentation is added/updated where necessary
- [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (below reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by mraible** [comment]
Maybe @michael-simons can help with this one?

**Comment by michael-simons** [comment]
What exactly looks broken?

**Comment by mshima** [comment]
The sample app is failing to start due to:
```
2022-12-03T09:46:16.181Z  WARN 1 --- [o4jDriverIO-3-2] o.s.b.a.n.Neo4jReactiveHealthIndicator   : Health check failed 
[138](https://github.com/jhipster/generator-jhipster/actions/runs/3626752894/jobs/6116682773#step:27:140)
  
[139](https://github.com/jhipster/generator-jhipster/actions/runs/3626752894/jobs/6116682773#step:27:141)
  java.util.NoSuchElementException: Source was empty
[140](https://github.com/jhipster/generator-jhipster/actions/runs/3626752894/jobs/6116682773#step:27:142)
  	at reactor.core.publisher.MonoSingle$SingleSubscriber.onComplete(MonoSingle.java:174)
[141](https://github.com/jhipster/generator-jhipster/actions/runs/3626752894/jobs/6116682773#step:27:143)
  	at reactor.core.publisher.FluxMap$MapSubscriber.onComplete(FluxMap.java:144)
[142](https://github.com/jhipster/generator-jhipster/actions/runs/3626752894/jobs/6116682773#step:27:144)
  	at reactor.core.publisher.MonoFlatMapMany$FlatMapManyInner.onComplete(MonoFlatMapMany.java:260)
[143](https://github.com/jhipster/generator-jhipster/actions/runs/3626752894/jobs/6116682773#step:27:145)
  	at reactor.core.publisher.FluxZip$ZipCoordinator.drain(FluxZip.java:890)
[144](https://github.com/jhipster/generator-jhipster/actions/runs/3626752894/jobs/6116682773#step:27:146)
  	at reactor.core.publisher.FluxZip$ZipInner.onComplete(FluxZip.java:1142)
[145](https://github.com/jhipster/generator-jhipster/actions/runs/3626752894/jobs/6116682773#step:27:147)
  	at reactor.core.publisher.FluxCreate$BaseSink.complete(FluxCreate.java:460)
[146](https://github.com/jhipster/generator-jhipster/actions/runs/3626752894/jobs/6116682773#step:27:148)
  	at reactor.core.publisher.FluxCreate$SerializedFluxSink.drainLoop(FluxCreate.java:247)
[147](https://github.com/jhipster/generator-jhipster/actions/runs/3626752894/jobs/6116682773#step:27:149)
  	at reactor.core.publisher.FluxCreate$SerializedFluxSink.drain(FluxCreate.java:213)
[148](https://github.com/jhipster/generator-jhipster/actions/runs/3626752894/jobs/6116682773#step:27:150)
```

**Comment by mraible** [comment]
I can confirm that reactive JHipster apps with Spring Boot 3 don't start with the default `management.health.neo4j.enabled=true`. If I change it to `false`, the app starts.

File: generators/server/templates/src/main/resources/config/application.yml.ejs [hunk 1]
```diff
@@ -118,6 +118,10 @@ management:
   health:
     mail:
       enabled: false # When using the MailService, configure an SMTP server and set this to true
+<%_ if (databaseTypeNeo4j && reactive) { _%>
+    neo4j:
+      enabled: false
```
Comment (mraible) [inline generators/server/templates/src/main/resources/config/application.yml.ejs:123]:
```suggestion
      enabled: false # The health check for Neo4j is broken for Spring Boot 3
```

**Review by mraible — APPROVED** [review]

File: generators/server/templates/src/main/resources/config/application.yml.ejs [hunk 2]
```diff
@@ -118,6 +118,10 @@ management:
   health:
     mail:
       enabled: false # When using the MailService, configure an SMTP server and set this to true
+<%_ if (databaseTypeNeo4j && reactive) { _%>
+    neo4j:
+      enabled: false # The health check for Neo4j doesn't work with Spring Boot 3
```
Comment (mraible) [inline generators/server/templates/src/main/resources/config/application.yml.ejs:123]:
```suggestion
      enabled: false # The health check for Neo4j doesn't work with WebFlux
```

**Comment by mraible** [comment]
I experienced this today with a JHipster 7.9.3 app, so it's not something caused by Spring Boot 3. I'm gonna merge this PR.

**Pull request merged**